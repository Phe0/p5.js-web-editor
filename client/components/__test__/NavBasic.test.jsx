import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NavBasic from '../NavBasic';

describe('NavBasic', () => {
  let wrapper = null;
  const props = {
    onBack: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<NavBasic {...props} />);
  });

  it('it renders nav', () => {
    expect(wrapper.exists('.nav')).toEqual(true);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<NavBasic {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
