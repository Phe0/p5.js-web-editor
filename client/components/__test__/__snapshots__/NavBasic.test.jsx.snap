// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBasic renders correctly 1`] = `
<nav
  className="nav"
  title="main-navigation"
>
  <ul
    className="nav__items-left"
  >
    <li
      className="nav__item-logo"
    >
      <test-file-stub
        aria-label="p5.js Logo"
        className="svg__logo"
        focusable="false"
        role="img"
      />
    </li>
    <li
      className="nav__item"
    >
      <button
        onClick={[MockFunction]}
      >
        <span
          className="nav__item-header"
        >
          <test-file-stub
            aria-hidden="true"
            focusable="false"
          />
        </span>
        Back to the editor
      </button>
    </li>
  </ul>
</nav>
`;
